### PR TITLE
Add Enroll endpoint

### PR DESCRIPTION
**Pull request by J0**
## What kind of change does this PR introduce?

Add endpoint to enroll

## What is the current behavior?

Can't enroll a device

## What is the new behavior?

Feel free to include screenshots if it includes visual changes.

## Additional context

None for the moment

### CODE REVIEW

File: api/mfa.go [hunk 1]
```diff
@@ -101,3 +127,71 @@ func (a *API) GenerateRecoveryCodes(w http.ResponseWriter, r *http.Request) erro
 		RecoveryCodes: recoveryCodes,
 	})
 }
+
+func (a *API) EnrollFactor(w http.ResponseWriter, r *http.Request) error {
+	const FACTOR_PREFIX = "factor"
```
Comment (J0) [inline api/mfa.go:132]: rename this to camelCase

File: api/mfa.go [hunk 2]
```diff
@@ -101,3 +125,62 @@ func (a *API) GenerateRecoveryCodes(w http.ResponseWriter, r *http.Request) erro
...
+	}
+	params := &EnrollFactorParams{}
+	jsonDecoder := json.NewDecoder(r.Body)
+	err := jsonDecoder.Decode(params)
+	if err != nil {
+		return badRequestError("Could not read EnrollFactor params: %v", err)
+	}
+	if (params.FactorType != "totp") && (params.FactorType != "webauthn") {
+		return unprocessableEntityError("FactorType needs to be either 'totp' or 'webauthn'")
+	}
```
Comment (kangmingtay) [inline api/mfa.go:146]: i guess the implementation provided below supports `totp` for now only right? i'm not too sure if the `webauthn` provider follows the same enrollment process as `totp` but if so, then we can abstract this out into an interface where you have an `MFAProvider` that implements those generic methods. For example, something like:

```golang
type MFAProvider interface {
  Generate(...) (*EnrollFactorResponse, error)
  Verify(...) (*VerifyChallengeResponse, error)
}
```

**Review by kangmingtay — APPROVED** [review]

File: api/mfa.go (hunk 2 above)
Comment (J0) [reply api/mfa.go:146]: Yeah I think this is a good idea -- it's similar but not exactly the same

Comment (J0) [reply api/mfa.go:146]: Will add this in when I add webauthn in

**Pull request merged**